demos/mantle_convection/dynamic_topography/dynamic_topography.py

```diff
@@ -0,0 +1,236 @@
+# plotter = pv.Plotter(notebook=True)
+# plotter.add_mesh(temp_data)
+# plotter.camera_position = "xy"
+# plotter.show(jupyter_backend="static", interactive=False)
+# -
+
+# We next specify the constants and physical approximation for this problem.
+# Note that this case is time independent and hence, when compared to most of our previous
+# tutorials, no timestepping options are specified.
+
+Ra = Constant(1e4)
+approximation = BoussinesqApproximation(Ra)
+
+# As noted previously, with a free-slip boundary condition on both boundaries, one can add an arbitrary constant to the pressure solution (i.e. this case incorporates a pressure nullspace). These lead to null-modes (eigenvectors) for the linear system, resulting in a singular matrix.
+# In preconditioned Krylov methods these null-modes must be subtracted from the approximate solution at every iteration. We do that below,
+# setting up a nullspace object as we did in the previous tutorial.
+
+Z_nullspace = create_stokes_nullspace(Z, closed=True, rotational=False)
+
+# Boundary conditions are next specified. For velocity, we specify free‐slip conditions on both boundaries. As noted in our
+# 2-D cylindrical tutorial, we incorporate these <b>weakly</b> through the <i>Nitsche</i> approximation. Given we do not solve
+# an energy equation for this time independent case, no boundary conditions are required for temperature.
+
+stokes_bcs = {
+    boundaries.bottom: {'uy': 0},
+    boundaries.top: {'uy': 0},
+    boundaries.left: {'ux': 0},
+    boundaries.right: {'ux': 0},
+}
+
+# We can now setup and solve the variational problem for the Stokes equations,
+# passing in the approximation, nullspace and near-nullspace information configured above.
+
+# +
+stokes_solver = StokesSolver(
+    z,
+    T,
+    approximation,
+    bcs=stokes_bcs,
+    nullspace=Z_nullspace,
+    transpose_nullspace=Z_nullspace,
+    solver_parameters="direct",
+)
+
+stokes_solver.solve()
+# -
+
+# At this point, we have a global solution for velocity and pressure. Let's visualise them:
+
+# + tags=["active-ipynb"]
+# VTKFile("velocity.pvd").write(z.subfunctions[0])
+# u_data = pv.read("velocity/velocity_0.vtu").glyph(factor=1e-3, orient="Velocity", tolerance=1e-2)
+# plotter = pv.Plotter(notebook=True)
+# plotter.add_mesh(temp_data)
+# plotter.add_mesh(u_data, color="black")
+# plotter.camera_position = "xy"
+# plotter.show(jupyter_backend="static", interactive=False)
+# -
+
+# Next, the global solutions is used to compute normal stresses at both
+# top and bottom boundaries, via the `force_on_boundary` method.
+
+# +
+ns_top = stokes_solver.force_on_boundary(boundaries.top)
+ns_bottom = stokes_solver.force_on_boundary(boundaries.bottom)
+# -
+
+# With these normal stresses, we can now calculate the dynamic topography at both the surface and the core-mantle boundary (CMB) using:
+#
+# $$\delta_{h} = \frac{\sigma_{rr}}{(\Delta \rho * g)}$$
+#
+# where:
+#   - $\sigma_{rr}$ is the normal stress at the boundary,
```

Comment:
Might need updating if you decide to use `\sigma_{yy}` or `\sigma_{zz}` instead of `\sigma_{rr}`.